PHPCompatibility/Tests/Syntax/NewInterpolatedStringDereferencingUnitTest.inc

```diff
@@ -0,0 +1,27 @@
+<?php
+
+// OK: strings that don't use interpolation or dereferencing.
+$a = 'foo';
+$a = "foo";
+$a = "foo
+bar";
+
+// OK: strings that use interpolation, but not dereferencing.
+$a = "$bar";
+$a = "foo$bar";
+$a = "foo
+$bar";
+
+// OK: strings that use dereferencing, but not interpolation.
+$a = 'foo'[0];
+$a = "foo"[0];
+$a = 'foo'->baz();
+$a = "foo"->baz();
+$a = "foo
+bar"->baz();
+
+// PHP 8.0: strings that use interpolation and dereferencing.
+$a = "foo$bar"[0];
+$a = "foo$bar"->baz();
+$a = "foo
+$bar"->baz();
```

Comment:
I think we need one more test case here: a multi line text string which has variable interpolation on multiple lines to confirm only one error is thrown for this. With the current sniff code, the case will work correctly.
```
$a = "foo $lines
$bar
more foo"[10];
```